workers/unity/Assets/Playground/Scripts/EntityInitialization/GameObjectInitializationSystem.cs

```diff
@@ -84,11 +84,14 @@ protected override void OnUpdate()
                         spatialEntityId);
                 var gameObjectReference = new GameObjectReference { GameObject = gameObject };
 
+                var requiresSpatialOSBehaviourManagerComponent = new RequiresSpatialOSBehaviourManager();
+
                 entityGameObjectCache[entity.Index] = gameObject;
                 var gameObjectReferenceHandleComponent = new GameObjectReferenceHandle();
 
-                PostUpdateCommands.AddComponent(addedEntitiesData.Entities[i], gameObjectReferenceHandleComponent);
                 viewCommandBuffer.AddComponent(entity, gameObjectReference);
+                PostUpdateCommands.AddComponent(addedEntitiesData.Entities[i], requiresSpatialOSBehaviourManagerComponent);
```

Comment:
Yeah, that would be less error prone indeed but rather than having `NeedSomeSetupAndDeleteTheTagAfterwards` component tags, we have `AddMeAndKeppMeForeverToShowThatSetupIsNotRequiredAnymore` tags which bloat up the entity over time. Let me think about this.

Reply:
Yes I'm unsure of the costs of long-persisting components on entities